cli/src/cli_util.rs

```diff
@@ -633,6 +633,16 @@ impl CommandHelper {
         &self.matches
     }
 
+    pub fn subcommands(&self) -> impl Iterator<Item = &'_ str> {
+        let mut next = self.matches().subcommand();
+        
+        std::iter::from_fn(move || {
+            let (sub, args) = next.take()?;
+            next = args.subcommand();
+            Some(sub)
+        })
+    }
```

Comment:
Oh, neat - I think I knew about this at one time, but forgot about it.

lol, and the source even has a comment alluding to this exact implementation of it :joy: 
```rust
pub fn successors<T, F>(first: Option<T>, succ: F) -> Successors<T, F>
where
    F: FnMut(&T) -> Option<T>,
{
    // If this function returned `impl Iterator<Item=T>`
    // it could be based on `from_fn` and not need a dedicated type.
    // However having a named `Successors<T, F>` type allows it to be `Clone` when `T` and `F` are.
    Successors { next: first, succ }
}
```

I'll switch to that and likely forget about it and get to rediscover it all over again in the future :+1: